### PR TITLE
initial support for vor v7

### PR DESCRIPTION
**Pull request by lsh-0**
- [x] review

### CODE REVIEW

File: api-raml.sha1 [hunk 1]
```diff
@@ -1 +1 @@
-b0fcfa2b76b61b71ab830bcd41929f1758375851
+4d2280b937793aee9e55fc2b87a313dcc6e0481e
```
Comment (lsh-0) [inline api-raml.sha1:1]: will need to be replaced with actual sha

File: bot-lax-adaptor.sha1 [hunk 2]
```diff
@@ -1 +1 @@
-d0da160fd6f289d45f266e6e311f0861351a9224
+961c57c524b0b72315961602fbc9ba7ff874f048
```
Comment (lsh-0) [inline bot-lax-adaptor.sha1:1]: it doesn't look like elife-tools has support for the new data (`editorEvaluation`, `elifeAssessment`, `recommendationForAuthors`, `publicReviews` etc) that the api-raml change captures, so the bot-lax revision will have to stay the same essentially except for a bumped api-raml revision.

File: src/publisher/middleware.py [hunk 3]
```diff
@@ -164,18 +164,15 @@ def middleware(request):
...
 
-def vor_valid_under_v5(ajson):
-    "returns True if given article-json is valid under version 5 of the VOR spec."
-    # True, when an authorResponse is *not* present, or,
-    # when an authorResponse *is* present *and* includes a decisionLetter.
-    if "authorResponse" in ajson:
-        return "decisionLetter" in ajson
+def vor_valid_under_v6(ajson):
+    "returns True if given article-json is valid under version 6 of the VOR spec."
+    # TODO: pending outcome of https://github.com/elifesciences/api-raml/pull/292
```
Comment (lsh-0) [inline src/publisher/middleware.py:169]: todo

**Comment by lsh-0** [comment]
fyi @nlisgo , @gnott , lax changes above should put the vor v7 bump here https://github.com/elifesciences/api-raml/pull/292 into play *but only once it's merged*. Once merged, assuming that PR doesn't have any significant changes, this PR will need some tweaks but that's about it.

More work in elife-tools and bot-lax will be needed to support the new data described in that PR but changes to those projects will trigger automatic updates in lax.

@gnott , let me know if you want a hand with elife-tools/bot-lax. I'm in for a few hours tomorrow morning but otherwise not until Friday.

I have no clue when all this is due and no ticket to track this work but there was this in a mail out today:

Screenshot at 2023-01-25 15-28-30

which says Jan 31st? Obviously the journal is going to need to more than just support for an upgraded mime type if any PRC content is to be viewable downstream.

File: bot-lax-adaptor.sha1 (hunk 2 above)
Comment (lsh-0) [reply bot-lax-adaptor.sha1:1]: @gnott , I'm sure you're familiar with all these, they're more for my benefit:

recommendation for authors:
https://github.com/elifesciences/XML-mapping/blob/a559aaa70ac6239528e9ecabe0d2e69802337a95/elife-1234567890-v2/elife-1234567890-v2.xml#L2396-L2454

public reviews:
https://github.com/elifesciences/XML-mapping/blob/a559aaa70ac6239528e9ecabe0d2e69802337a95/elife-1234567890-v2/elife-1234567890-v2.xml#L2046-L2395

editor's assessment:
https://github.com/elifesciences/XML-mapping/blob/a559aaa70ac6239528e9ecabe0d2e69802337a95/elife-1234567890-v2/elife-1234567890-v2.xml#L2001-L2045

**Comment by gnott** [comment]
Yes, I'll be doing some `elife-tools` parsing function today and I will probably reach completion and release a new version since there will be no backwards-compatibility problems.

**Comment by lsh-0** [comment]
> since there will be no backwards-compatibility problems.

This is my hope too, it will make things easier for lax content negotiation/downgrading. I've been caught out before thinking no changes were required so I'm keeping my ear to the ground.

**Comment by gnott** [comment]
To give a progress update on `elife-tools`, I have feature complete code with good tests. Nothing is committed yet because I'd like to look for possible refactoring and add comments and things. I expect a new release this week for sure. When it's ready, I can draft a `bot-lax-adaptor` PR for review which will invoke the new parsing functions.

**Comment by lsh-0** [comment]
> Nothing is committed yet because I'd like to look for possible refactoring and add comments and things.

No worries. I've taken to leaving a day between finishing code and then going back and reviewing it with fresh eyes. 

I'd be happy to take a look at anything you've done in a draft PR with the understanding that it's not complete yet.

Let me know if I can help prepare bot-lax at all.

**Comment by gnott** [comment]
Draft `bot-lax-adaptor` PR for viewing https://github.com/elifesciences/bot-lax-adaptor/pull/918

**Comment by gnott** [comment]
To @lsh-0, I don't know if PR https://github.com/elifesciences/lax/pull/835 will pass its tests as compared to the work you've prepared on this PR. May I suggest if you use the sha for the `api-raml` and `bot-lax-adaptor` from that PR then it should make the vor v7 schema available for using in Lax.

**Comment by lsh-0** [comment]
yeah, I'll clean all this up today, investigate these build failures and get it deployed

**Comment by nlisgo** [comment]
This looks encouraging. @FAtherden-eLife is best placed to advise on timelines for the first articles which will be published which will contain these new sections.

It is my understanding that the soonest could be within a week of the decision to progress the first reviewed-preprint to Version of Record. In theory, that decision could come in any day. For this reason, we have progressed the work on api-dummy, api-sdk-php and journal so that we can have high confidence that when lax is ready we can switch to v7.

**Comment by lsh-0** [comment]
> lax is ready we can switch to v7.

lax, bot-lax and elife-tools are done and deployed

**Pull request merged**